botocore/endpoint.py

```diff
@@ -234,41 +248,39 @@ def create_endpoint(self, service_model, region_name=None, is_secure=True,
                         response_parser_factory=None, timeout=DEFAULT_TIMEOUT):
         if region_name is None:
             region_name = self._configured_region
-        # Use the endpoint resolver heuristics to build the endpoint url.
-        scheme = 'https' if is_secure else 'http'
-        try:
-            endpoint = self._endpoint_resolver.construct_endpoint(
-                service_model.endpoint_prefix,
-                region_name, scheme=scheme)
-        except BaseEndpointResolverError:
-            if endpoint_url is not None:
-                # If the user provides an endpoint_url, it's ok
-                # if the heuristics didn't find anything.  We use the
-                # user provided endpoint_url.
-                endpoint = {'uri': endpoint_url, 'properties': {}}
-            else:
-                raise
-
         if endpoint_url is not None:
-            # If the user provides an endpoint url, we'll use that
-            # instead of what the heuristics rule gives us.
+            # If the user provides an endpoint url, we'll it as-is.
             final_endpoint_url = endpoint_url
         else:
-            final_endpoint_url = endpoint['uri']
+            # Construct a URI from the scheme and hostname. This will
+            # only resolve and endpoint. It won't override signature
+            # versions, signing names, etc. That logic is all in
+            # ClientCreator. Note that ClientCreator will inject an
+            # endpoint_url into create_endpoint(), so this branch is
+            # typically not invoked unless used directly.
+            final_endpoint_url = self._get_endpoint_from_resolver(
```

Comment:
I kept this here for backwards compatibility, but I was wondering if we should refactor this slightly to require an endpoint_url and region, given that endpoint resolution occurs in ClientCreator (eliminating this conditional branch and the need for the EndpointCreator to utilize an endpoint resolver). Is this class deemed part of the public API (meaning we should not change this aspect of it), or would it be ok to make this change considering endpoint_url is always injected from ClientCreator?